src/features/groups/components/InviteMemberDialog/InviteMemberDialogContent.tsx

```diff
@@ -30,23 +30,42 @@ export const InviteMemberDialogContent = ({
   groupId,
 }: IInviteMemberDialogContentProps) => {
   const [isLinkButtonClicked, setIsLinkButtonClicked] = useState(false);
+  const [isHash, setHash] = useState('');
 
+  /**
+   * This useEffect is fired when the `isDialogOpen` state is changed(true/false). if true, the setup is fired.
+   * generating the invitation link hash and put it as state `isHash`
+   */
+  useEffect(() => {
+    const getHash = async () => {
+      if (isDialogOpen) {
+        const result = await putGenerateInvitationLinkHash(groupId);
+        if (!result.ok) {
+          alert('Something went wrong. Please try again');
+          return;
+        }
+        setHash(result.value.invitationLinkHash);
+      }
+    };
+    getHash();
+  }, [isDialogOpen, groupId]);
   /**
    * Handle the link copy button click.
-   * If the generatedLink is not valid, do nothing.
    * If success, generated invitation link URL is copied to clipboard and change the text of button to 'Copied!'
+   * *hash is already generated when the dialog open
    */
   const handleLinkCopy = async () => {
     setIsLinkButtonClicked(true);
-    const result = await putGenerateInvitationLinkHash(groupId);
-
-    if (!result.ok) {
-      alert('Something went wrong. Please try again');
-      return;
+    try {
+      return await navigator.clipboard.writeText(`${CLIENT_BASE_URL}/groups/join/${isHash}`);
```

Comment:
```
if (!hash) throw new Error('Invitation link hash is not set.');
return await navigator.clipboard.writeText(`${CLIENT_BASE_URL}/groups/join/${isHash}`);
```

It would be better to throw an error if the hash is not set, so the execution will go to the `catch` block instead of becoming successful.